perfkitbenchmarker/providers/yandexcloud/yc_virtual_machine.py

```diff
@@ -0,0 +1,475 @@
+
+  @classmethod
+  def _GetOptionDecoderConstructions(cls):
+    """Gets decoder classes and constructor args for each configurable option.
+
+    Returns:
+      dict. Maps option name string to a (ConfigOptionDecoder class, dict) pair.
+          The pair specifies a decoder class and its __init__() keyword
+          arguments to construct in order to decode the named option.
+    """
+    result = super(YcVmSpec, cls)._GetOptionDecoderConstructions()
+    result.update({
+        'preemptible': (option_decoders.BooleanDecoder, {'default': False}),
+        'boot_disk_size': (option_decoders.IntDecoder, {'default': None}),
+        'boot_disk_type': (option_decoders.StringDecoder, {'default': None}),
+        'folder_id': (option_decoders.StringDecoder, {'default': None}),
+        'image_family': (option_decoders.StringDecoder, {'default': None}),
+        'image_folder_id': (option_decoders.StringDecoder, {'default': None}),
+        'core_fraction': (option_decoders.IntDecoder, {'default': None}),
+        'platform_id': (option_decoders.StringDecoder, {'default': None}),
+    })
+    return result
+
+
+class YcVirtualMachine(virtual_machine.BaseVirtualMachine):
+  """Object representing a Yandex Cloud Virtual Machine."""
+
+  CLOUD = providers.YC
+  # Subclasses should override the default image OR
+  # both the image family and image_folder_id.
+  DEFAULT_IMAGE = None
+  DEFAULT_IMAGE_FAMILY = None
+  DEFAULT_IMAGE_FOLDER_ID = "standard-images"
+  DEFAULT_USER_NAME = "yc-user"
+  BOOT_DISK_SIZE_GB = 10
+  BOOT_DISK_TYPE = yc_disk.NET_HDD
+
+  _host_lock = threading.Lock()
+  deleted_hosts = set()
+  host_map = collections.defaultdict(list)
+
+  def __init__(self, vm_spec):
+    """Initialize a YC virtual machine.
+
+    Args:
+      vm_spec: virtual_machine.BaseVmSpec object of the vm.
+
+    Raises:
+      errors.Config.MissingOption: If the spec does not include both "cores" and "memory"
+    """
+    super(YcVirtualMachine, self).__init__(vm_spec)
+    self.user_name = self.DEFAULT_USER_NAME
+    self.boot_metadata = {}
+    self.image = self.image or self.DEFAULT_IMAGE
+    self.preemptible = vm_spec.preemptible
+    self.core_fraction = vm_spec.core_fraction
+    self.platform_id = vm_spec.platform_id
+    self.machine_type = vm_spec.machine_type
+    self.memory, self.cores = util.ReturnFlavor(self.machine_type)
+    self.early_termination = False
+    self.preemptible_status_code = None
+    self.folder_id = vm_spec.folder_id or util.GetDefaultFolderId()
+    self.image_family = vm_spec.image_family or self.DEFAULT_IMAGE_FAMILY
+    self.image_folder_id = vm_spec.image_folder_id or self.DEFAULT_IMAGE_FOLDER_ID
+    self.backfill_image = False
+    self.network = yc_network.YcNetwork.GetNetwork(self)
+    self.boot_disk_size = vm_spec.boot_disk_size
+    self.boot_disk_type = vm_spec.boot_disk_type
+    self.id = None
+
+    if not self.cores or not self.memory:
+        raise errors.Config.MissingOption
```

Comment:
Can you put a message in here like `raise errors.Config.MissingOption('Missing cores and/or memory')`